mitodl/odl-video-service
# Youtube API integration

#### What are the relevant tickets?
Closes #343 
Closes #347 
Closes #349 

#### What's this PR do?
If ``settings.ENABLE_VIDEO_PERMISSIONS``:
  - Do not allow video permissions to be public unless a subtitle exists
  - Remove public video permissions if the subtitle is deleted
  - If the video is made public, upload it and its subtitle to Youtube
  - If the video subtitle is changed, upload the new one to Youtube
  - If the video is deleted or made non-public, delete it from Youtube

#### How should this be manually tested?
- Follow the steps in the README to set up Youtube env variables with your personal Youtube account.
- Check that the 'Public' video permission for videos is disabled unless a subtitle exists.
- Make a video public and save it.
- Go to https://www.youtube.com/my_videos, your video should show up here eventually, along with its captions
- Delete the video's subtitle from OVS. The video's permissions should be reset back to default collection permissions and the video should be deleted from Youtube.
- Add the subtitles back, and make the video public again.  Wait for it to show up in Youtube.
- Change the video permissions to anything besides public and save it.  The Youtube version of the video should be deleted.
